pkg/reconciler/v1alpha1/autoscaling/kpa/kpa.go

```diff
@@ -214,24 +197,85 @@ func (c *Reconciler) reconcile(ctx context.Context, pa *pav1alpha1.PodAutoscaler
 	}
 
 	// Compare the desired and observed resources to determine our situation.
-	got := 0
+	got, err := c.addressCount(ctx, pa)
+	if err != nil {
+		return err
+	}
+
+	logger.Infof("PA got=%v, want=%v", got, want)
+
+	err = c.setupReporter(pa, got, want)
+	if err != nil {
+		return err
+	}
+
+	updatePAStatus(pa, got, want)
+	return nil
+}
+
+func (c *Reconciler) updateStatus(desired *pav1alpha1.PodAutoscaler) (*pav1alpha1.PodAutoscaler, error) {
+	pa, err := c.paLister.PodAutoscalers(desired.Namespace).Get(desired.Name)
+	if err != nil {
+		return nil, err
+	}
+	// If there's nothing to update, just return.
+	if reflect.DeepEqual(pa.Status, desired.Status) {
+		return pa, nil
+	}
+	// Don't modify the informers copy
+	existing := pa.DeepCopy()
+	existing.Status = desired.Status
+
+	return c.ServingClientSet.AutoscalingV1alpha1().PodAutoscalers(pa.Namespace).UpdateStatus(existing)
+}
 
+func (c *Reconciler) addressCount(ctx context.Context, pa *pav1alpha1.PodAutoscaler) (int, error) {
+	logger := logging.FromContext(ctx)
+	got := 0
 	// Look up the Endpoints resource to determine the available resources.
 	endpoints, err := c.endpointsLister.Endpoints(pa.Namespace).Get(pa.Spec.ServiceName)
 	if errors.IsNotFound(err) {
 		// Treat not found as zero endpoints, it either hasn't been created
 		// or it has been torn down.
 	} else if err != nil {
 		logger.Errorf("Error checking Endpoints %q: %v", pa.Spec.ServiceName, err)
-		return err
+		return 0, err
 	} else {
 		for _, es := range endpoints.Subsets {
 			got += len(es.Addresses)
 		}
 	}
+	return got, nil
+}
 
-	logger.Infof("PA got=%v, want=%v", got, want)
+func (c *Reconciler) kpaMetric(ctx context.Context, desiredMetric *autoscaler.Metric) (*autoscaler.Metric, error) {
+	logger := logging.FromContext(ctx)
+	metric, err := c.kpaMetrics.Get(ctx, desiredMetric.Namespace, desiredMetric.Name)
+	if errors.IsNotFound(err) {
+		metric, err = c.kpaMetrics.Create(ctx, desiredMetric)
+		if err != nil {
+			logger.Errorf("Error creating Metric: %v", err)
+			return nil, err
+		}
+	} else if err != nil {
+		logger.Errorf("Error fetching Metric: %v", err)
+		return nil, err
+	}
+
+	// Ignore status when reconciling
+	desiredMetric.Status = metric.Status
+	if !equality.Semantic.DeepEqual(desiredMetric, metric) {
+		metric, err = c.kpaMetrics.Update(ctx, desiredMetric)
+		if err != nil {
+			logger.Errorf("Error update Metric: %v", err)
+			return nil, err
+		}
+	}
+
+	return metric, nil
+}
 
+func (c *Reconciler) setupReporter(pa *pav1alpha1.PodAutoscaler, got int, want int32) error {
```

Comment:
This doesn't only setup the reporter but also reports the values we want. I'd rename this to `reportMetrics`? It creating a new reporter is an impl. detail.

Reply:
Done.